packages/eslint-plugin/docs/rules/class-literals-style.md

```diff
@@ -0,0 +1,96 @@
+# Ensures that literals on classes are exposed in a consistent style (`class-literals-style`)
+
+When writing TypeScript applications, it's typically safe to store literal values on classes using fields with the `readonly` modifier to prevent them from being reassigned.
+When writing TypeScript libraries that could be used by Javascript users however, it's typically safer to expose these literals using `getter`s, since the `readonly` modifier is enforced at compile type.
+
+## Rule Details
+
+This rule aims to ensure that literals exposed by classes are done so consistently, in one of the two style described above.
+Since both styles have their place, this rule by default does nothing - you must provide it with your desired style: either `fields` or `getters`.
+
+Note that this rule only checks for _literal_ values, and so not objects, arrays, or functions.
```

Comment:
nit clarification
```suggestion
Note that this rule only checks for constant _literal_ values (string, template string, number, bigint, boolean, regexp, null). It does not check objects or arrays, because a readonly field behaves differently to a getter in those cases. It also does not check functions, as it is a common pattern to use readonly fields with arrow function values as auto-bound methods.
```